portmapper/mapper.go

```diff
@@ -67,15 +70,14 @@ func (pm *PortMapper) Map(container net.Addr, hostIP net.IP, hostPort int, usePr
 
 // MapRange maps the specified container transport address to the host's network address and transport port range
 func (pm *PortMapper) MapRange(container net.Addr, hostIP net.IP, hostPortStart, hostPortEnd int, useProxy bool) (host net.Addr, err error) {
-	pm.lock.Lock()
-	defer pm.lock.Unlock()
-
 	var (
 		m                 *mapping
 		proto             string
 		allocatedHostPort int
 	)
-
+	lockKey := fmt.Sprintf("%s:%d-%d", hostIP, hostPortStart, hostPortEnd)
```

Comment:
You're right. It seems like it's impossible to do it correct with the current design :/